Src/IronPython/Runtime/Operations/PythonOps.cs

```diff
@@ -2422,7 +2422,7 @@ public static partial class PythonOps {
             return e;
         }
 
-        private static Exception MakeExceptionWorker(CodeContext/*!*/ context, object type, object value, object traceback, bool forRethrow) {
+        internal static Exception MakeExceptionWorker(CodeContext/*!*/ context, object type, object value, object traceback, bool forRethrow, bool forGenerator=false) {
```

Comment:
I think this is going to lead to some intellisense confusion (do I use MakeException or MakeExceptionWorker). Instead of exposing `MakeExceptionWorker` internally could we maybe add an (internal) `MakeExceptionForGeneratorTypeError` or something of the sort?